File: Examples/CSharp/Articles/RetrieveTaskEmbeddedDocuments.cs
namespace Aspose.Tasks.Examples.CSharp.Articles
{
    using System.IO;

    using Aspose.Words;
    using Aspose.Words.Drawing;

    internal class RetrieveTaskEmbeddedDocuments
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:RetrieveTaskEmbeddedDocuments
            //ExFor: Tsk.NotesRTF
            //ExFor: Shape
            //ExSummary: Shows how to extract embedded RTF and save OLE objects from it. 
            var project = new Project(dataDir + "TaskEmbeddedDocuments.mpp");
            var task = project.RootTask.Children.GetById(1);

            File.WriteAllText(dataDir + "Notes_out.rtf", task.Get(Tsk.NotesRTF));

            Document doc;
            using (var stream = new MemoryStream())
            {
                using (var writer = new StreamWriter(stream))
                {
                    writer.Write(task.Get(Tsk.NotesRTF));
                    doc = new Document(stream);
                }
            }

            var shapes = doc.GetChildNodes(NodeType.Shape, true);
            foreach (var node in shapes)
            {
                var shape = (Shape)node;
                if (shape.OleFormat == null 
                    || shape.OleFormat.IsLink
                    || shape.OleFormat.ProgId != "Word.Document.12")
                {
                    continue;
                }

                // Extract OLE Word object
                using (var stream = new MemoryStream())
                {
                    shape.OleFormat.Save(stream);

                    var newDocument = new Document(stream);
                    newDocument.Save(dataDir + "RetrieveTaskEmbeddedDocuments_out.doc");
                }
            }
            //ExEnd:RetrieveTaskEmbeddedDocuments
        }
    }
}

File: Examples/VisualBasic/Articles/RetrieveTaskEmbeddedDocuments.vb
Imports Aspose.Words
Imports System.IO

Namespace Articles
    Public Class RetrieveTaskEmbeddedDocuments
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' ExStart:RetrieveTaskEmbeddedDocuments
            Dim project As New Project(dataDir & "TaskEmbeddedDocuments.mpp")
            Dim task As Task = project.RootTask.Children.GetById(1)

            File.WriteAllText(dataDir & "Notes_out.rtf", task.[Get](Tsk.NotesRTF))

            Dim doc As Document = Nothing
            Using stream As New MemoryStream()
                Using streamWriter As New StreamWriter(stream)
                    streamWriter.Write(task.[Get](Tsk.NotesRTF))
                    doc = New Document(stream)
                End Using
            End Using

            Dim shapes As NodeCollection = doc.GetChildNodes(NodeType.Shape, True)
            For Each shape As Aspose.Words.Drawing.Shape In shapes
                If shape.OleFormat IsNot Nothing Then
                    If Not shape.OleFormat.IsLink Then
                        ' Extract OLE Word object
                        If shape.OleFormat.ProgId = "Word.Document.12" Then
                            Dim stream As New MemoryStream()
                            shape.OleFormat.Save(stream)

                            Dim newDoc As New Document(stream)
                            newDoc.Save(dataDir & "RetrieveTaskEmbeddedDocuments_out.doc")
                        End If
                    End If
                End If
            Next
            ' ExEnd:RetrieveTaskEmbeddedDocuments
        End Sub
    End Class
End Namespace